src/queryResult/sqlCodeLensProvider.ts

```diff
@@ -9,12 +9,22 @@ import ConnectionManager from "../controllers/connectionManager";
 import { QueryEditor } from "../constants/locConstants";
 import { generateDatabaseDisplayName, generateServerDisplayName } from "../models/connectionInfo";
 
-export class SqlCodeLensProvider implements vscode.CodeLensProvider {
-    constructor(private _connectionManager: ConnectionManager) {}
+export class SqlCodeLensProvider implements vscode.CodeLensProvider, vscode.Disposable {
+    private _disposables: vscode.Disposable[] = [];
+    private _codeLensChangedEmitter = new vscode.EventEmitter<void>();
+    public readonly onDidChangeCodeLenses?: vscode.Event<void> = this._codeLensChangedEmitter.event;
+
+    constructor(private _connectionManager: ConnectionManager) {
+        this._disposables.push(
+            this._connectionManager.onConnectionsChanged(() => {
```

Comment:
What connection changes from outside the editor are occuring that we want to impact the editor?  For example, if I open an editor from a specific connection, then edit the connection (using the connection dialog), I don't think we'd want to change that editor's already-established connection.  Are there other instances this is capturing?